src/test/java/org/apache/drillui/test/framework/pages/BasePage.java

```diff
@@ -69,4 +79,101 @@ protected static void sendText(WebElement element, String text) {
     element.sendKeys(Keys.CONTROL, "v");
   }
 
+  protected void waitForCondition(Function<WebDriver, Boolean> condition) {
+    waitForCondition(condition, TestProperties.getInt("DEFAULT_TIMEOUT"));
+  }
+
+  protected void waitForCondition(Function<WebDriver, Boolean> condition, int timeOut) {
+    if (!condition.apply(getDriver())) {
+      new WebDriverWait(getDriver(), timeOut)
+          .until(condition);
+    }
+  }
+
+  protected void waitForCondition(Function<WebDriver, Boolean> condition, Runnable workaroundAction) {
+    int timeStep = 1;
+    int elapsed = 0;
+    boolean success = false;
+    while (!condition.apply(getDriver()) && elapsed < TestProperties.getInt("DEFAULT_TIMEOUT") - 1) {
+      try {
+        waitForCondition(condition, timeStep);
+        success = true;
+      } catch (Exception e) {
+        workaroundAction.run();
+      }
+      elapsed += timeStep;
+    }
+    if (!success) {
+      waitForCondition(condition, timeStep);
+    }
+  }
+
+  protected boolean isElementStable(WebElement element) {
+    Point position = element.getLocation();
+    try {
+      Thread.sleep(100);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+    return position.equals(element.getLocation());
+  }
+
+  protected void forceClick(WebElement element) {
+    Actions actions = new Actions(getDriver());
+    int elapsed = 0;
+    int timeStep = 500;
+    boolean success = false;
+    while (elapsed < TestProperties.getInt("DEFAULT_TIMEOUT") * 1000) {
```

Comment:
You can also use Guava Stopwatch here to handle elapsed time.

Reply:
Done.